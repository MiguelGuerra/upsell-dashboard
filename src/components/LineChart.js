import React from 'react'
import { Line } from 'react-chartjs-2'

function LineChart({
    labelsArray,
    caption,
    dataArray,
    lineColor,
    backgroundColor,
    fillLine }) {
    const data = {
        labels: labelsArray,
        datasets: [
            {
                label: caption,
                data: dataArray,
                borderColor: lineColor,
                backgroundColor: backgroundColor,
                pointBackgroundColor: lineColor,
                pointBorderColor: lineColor,
                fill: fillLine
            }
        ]
    }

    return (
        <div data-testid="line-chart" className="line-chart">
            <Line data={data} />
        </div>
    )
}

export default LineChart
