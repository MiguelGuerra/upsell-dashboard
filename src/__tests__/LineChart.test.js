import React from 'react'
import renderer from 'react-test-renderer'
import LineChart from '../components/LineChart'

it("LineChart renders when is receives props", () => {
    const tree = renderer.create(
        <LineChart
            labelsArray={ ['test 1', 'test 2']}
            caption="Test Caption"
            dataArray={['45', '50']}
            lineColor="#41a6c3"
            backgroundColor="rgba(65, 166, 195, .5)"
            fillLine={true}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})