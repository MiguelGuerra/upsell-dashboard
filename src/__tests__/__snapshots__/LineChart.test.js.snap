// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart renders when is receives props 1`] = `
<div
  className="line-chart"
  data-testid="line-chart"
>
  <canvas
    data-testid="canvas"
    height={150}
    onClick={[Function]}
    role="img"
    width={300}
  />
</div>
`;
